site-scrapers/BostonMedicalCenter/index.js

```diff
@@ -0,0 +1,57 @@
+const { sites } = require("./config");
+const mychart = require("../../lib/MyChartAPI");
+
+module.exports = async function GetAvailableAppointments() {
+    console.log("BMC starting.");
+    const webData = await ScrapeWebsiteData();
+    console.log("BMC done.");
+    const results = [];
+    const timestamp = new Date();
+
+    for (const site of sites) {
+        results.push({
+            ...webData[site.departmentID],
+            ...site,
+            timestamp,
+        });
+    }
+    return results;
+};
+
+async function ScrapeWebsiteData() {
+    // We need to go through the flow and use a request verification token
+    const [
+        cookie,
+        verificationToken,
+    ] = await mychart.GetCookieAndVerificationToken(
+        "https://mychartscheduling.bmc.org/mychartscheduling/SignupAndSchedule/EmbeddedSchedule?id=10033319,10033364,10033367,10033370,10033373&dept=10098245,10098242,10098243,10098244,10098241&vt=2008&lang=en-US"
+    );
+    return mychart.AddFutureWeeks(
+        "mychartscheduling.bmc.org",
+        "/MyChartscheduling/OpenScheduling/OpenScheduling/GetOpeningsForProvider?noCache=0.4024598146273777",
```

Comment:
Does this `noCache` nonce need to vary every invocation to be meaningful?

Reply:
I'm not sure. If that's a commonly used pattern, I'm not familiar with it, so I'm not sure what it's supposed to do. I did notice that if you go to the scheduling site and look at the API calls in the Network tab, it shows several calls with various `noCache` values.

Reply:
Generally a parameter like `noCache` is used for "cachebusting," to prevent a service like Akamai or Cloudflare or whatever from caching the result of the query that needs to have fresh information. By using a unique URL every time, it bypasses the cache and goes through to the origin server every time.

It such situations, it doesn't really matter what the value is as long as it's relatively unique.
This feels like 
```
`nocache=${Math.random()$}`
```